arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -0,0 +1,211 @@
+            self.exit.wait(params.duration)
+
+        # Secondary block interrupt is via the KeyboardInterrupt exception.
+        # This enables running the plugin stand-alone without a workflow.
+        except (KeyboardInterrupt, SystemExit):
+            print("\nReceived keyboard interrupt; Stopping data collection.\n")
+            self.finished_early = True
+
+        # In either the case of a keyboard interrupt or a cancel signal, we need to
+        # send the SIGINT to the subprocess.
+        if self.finished_early:
+            proc.send_signal(2)
+
+        output, _ = proc.communicate()
+
+        # The output from the `rtla timerlat hist` command is columnar in three
+        # sections, plus headers. The first section is histogram data, which will
+        # display either two or three colums per CPU (three if user threads are
+        # enabled). The second section is latency statistics for each of the same
+        # columns as the histogram section. The third section is collapsed totals of
+        # the statistical data from all CPUs.
+
+        # For the first and second sections, we want to retain the row/columnt format
+        # in the output for easy recreation of the data as a table. For the third
+        # section, we want to return key:value pairs for each of the IRQ, Thr, and Usr
+        # total stats.
+
+        # $ sudo rtla timerlat hist -u -d 1 -c 1,2 -E 10 -b 10
+        # # RTLA timerlat histogram
+        # # Time unit is microseconds (us)
+        # # Duration:   0 00:00:02
+        # Index   IRQ-001   Thr-001   Usr-001   IRQ-002   Thr-002   Usr-002
+        # 0          1000       944       134       999       998       986
+        # 10            0        55       863         0         1        13
+        # 20            0         1         3         0         0         0
+        # over:         0         0         0         0         0         0
+        # count:     1000      1000      1000       999       999       999
+        # min:          0         3         4         0         2         3
+        # avg:          1         7        10         0         2         3
+        # max:          7        21        24         3        11        15
+        # ALL:        IRQ       Thr       Usr
+        # count:     1999      1999      1999
+        # min:          0         2         3
+        # avg:          0         4         6
+        # max:          7        21        24
+
+        time_unit = ""
+        col_headers = []
+        latency_hist = []
+        stats_per_col = []
+        accumulator = latency_hist
+        total_irq_latency = {}
+        total_thr_latency = {}
+        total_usr_latency = {}
+
+        is_time_unit = re.compile(r"# Time unit is (\w+)")
+        is_header = re.compile(r"Index")
+        is_digit = re.compile(r"\d")
+        is_summary = re.compile(r"ALL")
+
+        output_lines = output.splitlines()
+        line_num = 0
```

Comment:
Instead of dealing with line numbers as list offsets, you could consider using `output_lines.pop(0)` to remove the first entry on each iteration.

You'd need to handle an `IndexError` exception if you don't actually want to check that `len(output_lines)>0` first... but unless you get output radically different than you expect (in which case an unhandled exception is probably not the worst outcome), that's only an issue on "phase 3".

Reply:
`pop()` would be good, but I've recommended [using an _iterator_ instead](https://github.com/arcalot/arcaflow-plugin-rtla/pull/1#discussion_r1836951427), below.

Reply:
Alright, the idea of the `.pop()` made sense to me, but I'm getting really strange results every way that I try it. For what is gained, It's not worth me putting more time into it.